test/nn/conv/test_message_passing.py

```diff
@@ -154,7 +154,45 @@ def __init__(self):
         super().__init__(aggr='add')
 
     def forward(self, x: Tensor, edge_index: Adj) -> Tensor:
```

Comment:
I don't think JIT works with the way we are handling union types right now, you can see: https://discuss.pytorch.org/t/no-graph-attribute-when-use-torch-jit-overload-method/76128 but its possible that it can be supported now without the `overload_method` because union types are now supported.. maybe to be investigated in another PR.

Reply:
Are you sure `Union` is now supported by TorchScript? Currently, we are mapping `Union` to different `@overload` function headers.

Reply:
I haven't tried the `Union` types: But what I meant by the comment is that actually, the current approach doesn't work either. This test passes only because of JIT is not tested.

Reply:
Yes, for TorchScript to work, the final types needs to be resolved somewhere, e.g., we have a `forward` method of a GNN layer with `edge_index: Adj`, then the main `forward` method of the model needs to either set `edge_index` to `Tensor` or `SparseTensor`.